service/model/project.go

```diff
@@ -0,0 +1,160 @@
+package model
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	"github.com/google/uuid"
+)
+
+type ProjectRepository interface {
+	Create(ctx context.Context, p Project) (Project, error)
+	Read(ctx context.Context, id uuid.UUID) (Project, error)
+	ReadAll(ctx context.Context) ([]Project, error)
+	Update(ctx context.Context, p Project) (Project, error)
+	Delete(ctx context.Context, id uuid.UUID) error
+}
+
+var (
+	errProjectNameRequired = errors.New("project name is required")
+)
+
+type Project struct {
+	ID                 uuid.UUID
+	Name               string
+	SlackChannelID     string
+	ReleaseAlertConfig ReleaseAlertConfig
+	CreatedAt          time.Time
+	UpdatedAt          time.Time
+}
+
+type ProjectCreation struct {
+	Name           string
+	SlackChannelID string
+	RlsAlertCfg    ReleaseAlertConfig
+}
+
+type ProjectUpdate struct {
+	ProjectID         uuid.UUID
+	Name              *string
+	SlackChannelID    *string
+	RlsAlertCfgUpdate ReleaseAlertConfigUpdate
+}
+
+type ReleaseAlertConfig struct {
```

Comment:
`ReleaseAlertConfig` defines the information that will be included in a release alerts for a given project. 

I chose 'ReleaseAlert' over 'ReleaseMessage' as I believe it’s a more suitable name

Reply:
`Alert` sounds too negative. :)
Also agree that `message` is a bit too neutral and general.
How about `notification`?

Reply:
Good point, let's go with `ReleaseNotification`